### ISSUE TITLE
Cannot execute run_gaze_sequential.py to the end

### ISSUE THREAD

**Issue by ryoyomyamagishi** [Ryoyo-NV/Gaze-Analysis-System#2]
I tried to run this script without Kibana, on Jetson AGX Xavier with Jetpack 4.4 flashed by SDK Manager.
But the script cannot executed to the end, and I found following error.

Error: Failed to read a frame data because gst thread is done already
Error: Gst thread is dead 

The full log when executed the script is attached.
Could you tell me something wrong?

Thanks.

[log.txt](https://github.com/Ryoyo-NV/Gaze-Analysis-System/files/7655596/log.txt)

**Comment by ryoyomyamagishi** [comment]
I attempted to run this script again, on Jetson Nano B01 with Jetpack 4.4 and Deepstream 5.0 flashed by SDK Manager.
I installed some packages with designating version, but I got error.

I attached the full log during running this script and the log of history.
*I once mistaken forgetting option when I run the script, then I killed that process.

Thanks.
[log_211208.txt](https://github.com/Ryoyo-NV/Gaze-Analysis-System/files/7676720/log_211208.txt)
[history.txt](https://github.com/Ryoyo-NV/Gaze-Analysis-System/files/7676722/history.txt)

**Comment by takutoi** [comment]
Prebuild TensorRT engines in this repository such as "age_megaasian.onnx_b8_fp16.engine" or "gender_wiki.onnx_b8_fp16.engine" are built for Xavier GPU(compute capability 7.2).
So if you want to run on Nano(5.3) or TX2(6.2), you need to rebuild them on your device.
Please see https://github.com/Ryoyo-NV/Gaze-Analysis-System/tree/main/model about pretrained models.

**Comment by ryoyomyamagishi** [comment]
I understood that provided engine file made for GPU Compute Capability 7.2, thus this script cannot run at Jetson except Xavier series by default.

Thanks for your support.

**Comment by takutoi** [comment]
About your first comment, it is reported "Internal data stream error." in your log.txt.
Please run with video file (not camera) and show me the log messages.

**Comment by ryoyomyamagishi** [comment]
Thank you for your reply.

I tried to re-flash jetpack 4.4 to Jetson AGX Xavier, and run the script with sample video file (/opt/nvidia/deepstream/deepstream-5.0/samples/streams/sample_qHD.h264).
But I found error as attached log file.

I can see the log got this time is different with the log I uploaded at first on this issue, but see same "Internal data stream error" was shown.

Do you have any opinion for these phenomena?

Thanks.
[history_211212.txt](https://github.com/Ryoyo-NV/Gaze-Analysis-System/files/7698805/history_211212.txt)
[log_211212.txt](https://github.com/Ryoyo-NV/Gaze-Analysis-System/files/7698806/log_211212.txt)

**Comment by takutoi** [comment]
Sorry, this program cannot read a raw H.264 file. Please use a MP4 file.
In case of a webcam, please check your webcam spec using "v4l2-ctl --list-formats-ext", and specify the options --width, --height, --max-fps and --codec=mjpg as needed.

**Comment by ryoyomyamagishi** [comment]
Thank you for your replying.

I could run the script with mp4 file included DeepStream SDK's sample.

$ python3 run_gaze_sequential.py /opt/nvidia/deepstream/deepstream-5.0/samples/streams/sample_1080p_h264.mp4 -- codec h264

And also I could run this with USB camera, by limiting cam's spec level, such as resolution, and framerate.

$ python3 run_gaze_sequential.py --width 640 --height 480 --media v4l2 --max-fps 15 /dev/video0

I seemed the spec, 640 x480 15 fps, could run this script most appropriately under my circumstance, connecting Logicool C615 to AGX Xavier Developer Kit.

I could run this script both with USB camera and video file at Jetpack 4.4, so I solved all issue.
Thank you for your support.
